msamp/deepspeed/runtime/zero/fp8_stage_1_and_2.py

```diff
@@ -0,0 +1,992 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# Adapted from DeepSpeed (Copyright 2019 The Microsoft DeepSpeed Team)
+
+"""ZeRO Optimizer with MS-AMP support."""
+
+from itertools import chain
+
+import torch
+from torch._utils import _flatten_dense_tensors, _unflatten_dense_tensors
+from deepspeed import comm as dist
+
+from msamp.common.tensor import ScalingTensor, ScalingMeta
+from msamp.common.dtype import Dtypes
+from msamp.nn import model_state
+from msamp.common.utils import TransformerEngineWrapper
+
+from deepspeed.runtime.zero.stage_1_and_2 import all_gather_dp_groups, DeepSpeedZeroOptimizer, \
+    get_accelerator, move_to_cpu, logger, see_memory_usage
+
+SINGLE_PARTITION_OF_FP8_GROUPS = 'single_partition_of_fp8_groups'
+
+MASTER_WEIGHT_QTYPE = Dtypes.kfloat16
+WEIGHT_GRAD_QTYPE = Dtypes.kfloat8_e4m3
+WEIGHT_QTYPE = Dtypes.kfloat8_e4m3
```

Comment:
Will not change the weight type in the future?

Reply:
Yes. We will provide an API or new configuration to change them.